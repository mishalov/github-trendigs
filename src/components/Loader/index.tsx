import React from "react";
import "./Loader.scss";

/**
 * Loader, use if you want user to wait something
 */
const Loader: React.FC = props => (
  <div className="loader__wrapper">
    <div className="loader">
      <div className="loader-article">
        <div className="loader-page" />
      </div>
      <div className="loader__caption">{props.children}</div>
    </div>
  </div>
);

export default Loader;
