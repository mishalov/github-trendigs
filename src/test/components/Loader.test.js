import React from "react";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import Loader from "../../components/Loader";

Enzyme.configure({ adapter: new Adapter() });
const wrappers = [];

beforeEach(() => {
  wrappers.push(shallow(<Loader>test</Loader>));
});

describe("<Loader /> rendering", () => {
  it("renders correctly", () => {
    expect(wrappers[0]).toMatchSnapshot();
  });
});
