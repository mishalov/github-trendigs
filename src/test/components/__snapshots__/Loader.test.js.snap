// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> rendering renders correctly 1`] = `
<div
  className="loader__wrapper"
>
  <div
    className="loader"
  >
    <div
      className="loader-article"
    >
      <div
        className="loader-page"
      />
    </div>
    <div
      className="loader__caption"
    >
      test
    </div>
  </div>
</div>
`;
